annif/eval.py

```diff
@@ -145,10 +145,43 @@ def _evaluate_samples(self, y_true, y_pred, metrics='all'):
 
         return results
 
-    def results(self, metrics='all'):
+    def output_result_per_label(self, y_true, y_pred, results_file):
+        """Write results per label (non-aggregated) to outputfile"""
+
+        y_pred = y_pred.T > 0.0
+        y_true = y_true.T > 0.0
+
+        tp = (y_true & y_pred)
+        fp = (~y_true & y_pred)
+        fn = (y_true & ~y_pred)
+
+        r = len(y_true)
+
+        with open(results_file, 'w') as f:
+            f.write('\t'.join(['URI', 'Label', 'Support', 'True_positives',
+                               'False_positives', 'False_negatives',
+                               'Precision', 'Recall', 'F1_score']) + '\n')
+            zipped = zip(self._subject_index._uris,       # URI
+                         self._subject_index._labels,     # Label
+                         [sum(tp[i]) + sum(fn[i])
```

Comment:
I think it should be possible to use NumPy [ndarray.sum](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.sum.html#numpy.ndarray.sum) operations here, instead of using a for loop. This would be more efficient and less verbose

Reply:
Good point, I did this for the simple summations. For the precision and recall computations, I kept the list comprehensions in order to use sklearn metrics per row. Do you agree?

Reply:
Yes, fine to keep the for loops for those. Much better now!